drv/lpc55-update-server/src/images.rs

```diff
@@ -2,12 +2,31 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
+/// Implement LPC55_`update_server`'s knowledge about bits inside an image.
+///
+/// The update server needs to work with partial or otherwise invalid images.
+/// Signature checks are only performed at boot time. The update server
+/// does match FWIDs against the boot-time-info in some cases. But, because
+/// flash areas are mutated during update_server operations and stuff can
+/// happen, any data in the non-active Hubris image needs to be treated as
+/// untrusted (update_server does not alter its own image).
+/// Data structures, pointers, and offsets within an image are tested to
+/// ensure no mischief during `update_server` operations. The remainder
+/// of reliability and security concerns rely on the boot-time policies
+/// of the LPC55 ROM and the stage0 bootloader.
+use crate::{
+    indirect_flash_read, round_up_to_flash_page, SIZEOF_U32, U32_SIZE,
+};
+use abi::{ImageHeader, CABOOSE_MAGIC, HEADER_MAGIC};
 use core::ops::Range;
-use drv_lpc55_update_api::BLOCK_SIZE_BYTES;
-use drv_lpc55_update_api::{RotComponent, SlotId};
+use drv_lpc55_update_api::{RawCabooseError, RotComponent, SlotId};
 use drv_update_api::UpdateError;
-use userlib::UnwrapLite;
+use zerocopy::{AsBytes, FromBytes};
 
+// Our layout of flash banks on the LPC55.
+// Addresses are from the linker.
+// The bootloader (`bootleby`) resides at __STAGE0_BASE and
+// only cares about IMAGE_A and IMAGE_B.
 // We shouldn't actually dereference these. The types are not correct.
```

Comment:
Nit: after we switched these to zero-length arrays, dereferencing is actually safe and the types are arguably correct.